Content.Client/ADT/MesonVision/MesonVisionOverlay.cs

```diff
@@ -58,16 +59,27 @@ protected override void Draw(in OverlayDrawArgs args)
 
         foreach (var entity in _entity.GetEntities())
         {
-            if (_mob.HasComponent(entity) || _marker.HasComponent(entity)) continue;
-            if (!_spriteQuery.TryGetComponent(entity, out var sprite) || sprite.DrawDepth < -2 || sprite.DrawDepth > 7 || sprite.DrawDepth == 0) continue;
-            if (!_xformQuery.TryGetComponent(entity, out var xform)) continue;
-            if (_container.TryGetOuterContainer(entity, xform, out var container)) continue;
-
-            if (xform.MapID != mapId) continue;
+            // Ganimed edit start
+            if (!_spriteQuery.TryGetComponent(entity, out var sprite) ||
+                !_xformQuery.TryGetComponent(entity, out var xform) ||
+                xform.MapID != mapId ||
+                _container.TryGetOuterContainer(entity, xform, out var _))
+            {
+                continue;
+            }
+            // Ganimed edit end
 
             var worldPos = _xformSystem.GetWorldPosition(xform);
             if (!worldBounds.Contains(worldPos)) continue;
 
+            // Ganimed edit start
+            var isWall = sprite.BaseRSI?.Path.ToString().Contains("/Structures/Walls/") == true; // Я не очень уверен, а это вообще надо :/
```

Comment:
Вместо этого можно использовать проверку на компонент `SunShadowCast` или компонент `IsRoof`
```suggestion
            var isWall = _entity.HasComponent<SunShadowCast>(entity) || _entity.HasComponent<IsRoof>(entity);
```